src/sources/eventstoredb_metrics/types.rs

```diff
@@ -232,3 +237,52 @@ impl<'de> Visitor<'de> for DriveVisitor {
         Err(serde::de::Error::missing_field("<Drive path>"))
     }
 }
+
+// Can be either an integer or a string like 30%
+fn percent_or_integer<'de, D>(deserializer: D) -> Result<usize, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    struct PercentOrInteger;
+    static PERCENT_REGEX: Lazy<Regex> = Lazy::new(|| Regex::new(r"(\d+)%").unwrap());
+
+    impl<'de> Visitor<'de> for PercentOrInteger {
+        type Value = usize;
+
+        fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+            formatter.write_str("string or map")
+        }
+
+        fn visit_str<E>(self, value: &str) -> Result<Self::Value, E>
+        where
+            E: de::Error,
+        {
+            if let Some(caps) = PERCENT_REGEX.captures(value) {
```

Comment:
You could probably do this just as easily with basic string methods (strip trailing `%` and then parse the integer) without needing to introduce a lazy regex.

Reply:
Yeah, I did consider that but thought a regex would be a bit more robust and easy to reason about. Performance here isn't much of a concern since the scrape runs on an interval.